rclcpp/src/rclcpp/utilities.cpp

```diff
@@ -115,6 +95,48 @@ remove_ros_arguments(int argc, char const * const argv[])
     alloc.deallocate(nonros_argv, alloc.state);
   }
 
+  return return_arguments;
+}
+
+std::vector<std::string>
+init_and_remove_ros_arguments(
+  int argc,
+  char const * const argv[],
+  const InitOptions & init_options)
+{
+  init(argc, argv, init_options);
+
+  using rclcpp::contexts::get_global_default_context;
+  auto rcl_context = get_global_default_context()->get_rcl_context();
+  return _remove_ros_arguments(argv, &(rcl_context->global_arguments), rcl_get_default_allocator());
```

Comment:
leave this comment to share my understanding. Since context is initialized, it should get `global_arguments` from the context, not to call `rcl_parse_arguments` twice.